tests/kafkatest/utils/util.py

```diff
@@ -168,7 +168,7 @@ def validate_delivery(acked, consumed, idempotence_enabled=False, check_lost_dat
             success = False
 
     # Are there duplicates?
-    if len(set(consumed)) != len(consumed):
+    if verify_duplicate_data and len(set(consumed)) != len(consumed):
```

Comment:
What do you think about moving this check to the `if idempotence_enabled` line so that we don't fail the test but still print the duplicate record message to the log?